### ISSUE TITLE
build fails with >=bison-3.7

### ISSUE THREAD

**Issue by MeisterP** [GoldenCheetah/GoldenCheetah#3586]
Goldencheetah fails to compile after updating to >=bison-3.7 on Gentoo.
Bison 3.7 changes how header files are included [[1](https://lists.gnu.org/archive/html/info-gnu/2020-07/msg00006.html)][[2](https://www.gnu.org/software/bison/manual/html_node/_0025define-Summary.html)], in that instead of copying and inserting the contents of a file, the file itself is included (by default as '"basename.h"').

Gentoo Tracker bug is https://bugs.gentoo.org/737026
Probably the same issue on debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=966704

```
x86_64-pc-linux-gnu-g++ -c -march=native -O3 -pipe -Wno-unused-variable -Wno-sign-compare -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DGC_WANT_PYTHON -DGC_VIDEO_QT5 -DGC_WANT_ROBOT -DQXT_STATIC -DSIP_STATIC_MODULE -DGC_HAVE_PYTHON -DGC_HAVE_ICAL -DGC_HAVE_LIBUSB -DLIBUSB_V_1 -DGC_HAVE_SAMPLERATE -DGC_HAVE_OVERVIEW -DGC_HAVE_GENERIC -DQT_NO_DEBUG -DQT_SVG_LIB -DQT_MULTIMEDIAWIDGETS_LIB -DQT_WEBENGINEWIDGETS_LIB -DQT_PRINTSUPPORT_LIB -DQT_CHARTS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DQT_MULTIMEDIA_LIB -DQT_WEBENGINE_LIB -DQT_WEBENGINECORE_LIB -DQT_QUICK_LIB -DQT_GUI_LIB -DQT_XML_LIB -DQT_SQL_LIB -DQT_QMLMODELS_LIB -DQT_WEBCHANNEL_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_CONCURRENT_LIB -DQT_SERIALPORT_LIB -DQT_POSITIONING_LIB -DQT_BLUETOOTH_LIB -DQT_CORE_LIB -I. -IANT -ITrain -IFileIO -ICloud -ICharts -IMetrics -IGui -ICore -IPlanning -I../qwt/src -I../qxt/src -I../qtsolutions/json -I../qtsolutions/qwtcurve -I../lmfit -I../levmar -IPython -Itrue/include -Itrue/include -Itrue/include -I/usr/include/libusb-1.0 -I/usr/include/python3.7m -I/usr/include/qt5 -I/usr/include/qt5/QtSvg -I/usr/include/qt5/QtMultimediaWidgets -I/usr/include/qt5/QtWebEngineWidgets -I/usr/include/qt5/QtPrintSupport -I/usr/include/qt5/QtCharts -I/usr/include/qt5/QtOpenGL -I/usr/include/qt5/QtWidgets -I/usr/include/qt5/QtMultimedia -I/usr/include/qt5/QtWebEngine -I/usr/include/qt5/QtWebEngineCore -I/usr/include/qt5/QtQuick -I/usr/include/qt5/QtGui -I/usr/include/qt5/QtXml -I/usr/include/qt5/QtSql -I/usr/include/qt5/QtQmlModels -I/usr/include/qt5/QtWebChannel -I/usr/include/qt5/QtQml -I/usr/include/qt5/QtNetwork -I/usr/include/qt5/QtConcurrent -I/usr/include/qt5/QtSerialPort -I/usr/include/qt5/QtPositioning -I/usr/include/qt5/QtBluetooth -I/usr/include/qt5/QtCore -I. -I/usr/include/libdrm -I/usr/lib64/qt5/mkspecs/linux-g++ -o DataFilter_yacc.o DataFilter_yacc.cpp
DataFilter.tab.c:152:10: fatal error: DataFilter.tab.h: No such file or directory
compilation terminated.
make[1]: *** [Makefile:32199: DataFilter_yacc.o] Error 1
make[1]: Leaving directory '/var/tmp/portage/app-misc/goldencheetah-3.6_alpha_p20200905/work/GoldenCheetah-6164dd963f154ce9a05661167b23c6c06b2649dd/src'
make: *** [Makefile:91: sub-src-make_first-ordered] Error 2

```

[build.log](https://github.com/GoldenCheetah/GoldenCheetah/files/5178234/build.log)

**Comment by MeisterP** [comment]
Pull request with possible fix at https://github.com/GoldenCheetah/GoldenCheetah/pull/3590

**Comment by flokli** [comment]
Followup PR of #3590 that also fixes CI is at #3628.
